peps/pep-0759.rst

```diff
@@ -0,0 +1,472 @@
+``hashes``
+    This is a dictionary of the format described in :pep:`694`, used to capture both the
+    `hash algorithm and hex encoded digests`_ of the physical ``.whl`` file, with the same
+    constraints as proposed in that PEP.  Since these hashes are immutable once uploaded
+    to PyPI, they serve as a critical validation that the externally hosted wheel hasn't
+    been corrupted or compromised.
+
+The only effect of a ``.rim`` file is to change the download URL for the wheel artifact in
+both the HTML and JSON interfaces in the `simple repository API`_.  In the HTML page for a
+package release, the ``href`` attribute **MUST** be the value of the ``uri`` key,
+including a ``#<hashname>=<hashvalue>`` fragment.  this hash fragment **MUST** be in
+exactly the same format as described the :pep:`376` originated `signed wheel file format`_
+in the ``.dist-info/RECORD`` file. The exact same rules for selection of hash algorithm
+and encoding is used here.
+
+Similarly in the `JSON response`_ the ``url`` key pointing to the download file must be
+the value of the :ref:`uri <file-format>` key, and the ``hashes`` dictionary **MUST** be
+included with values populated from the ``hashes`` dictionary provided above.
+
+In all other respects, a compliant package index should treat ``.rim`` files the same as
+``.whl`` files, with some other minor exceptions as outlined below. For example, ``.rim``
+files can be `deleted <https://pypi.org/help/#deletion>`__ and yanked (:pep:`592`) just
+like any ``.whl`` file, with the exact same semantics (i.e. deletions are permanent). When
+a ``.rim`` is deleted, an index **MUST NOT** allow a matching ``.whl`` or ``.rim`` file to
+be (re-)uploaded.
+
+Externally hosted wheels **MUST** be available before the corresponding ``.rim`` file is
+uploaded to PyPI, otherwise a publishing race condition is introduced, although this
+requirement **MAY** be relaxed for ``.rim`` files uploaded to a :pep:`694` staged release.
+
+Wheels can override RIMs
+========================
+
+Indexes **MUST** reject ``.rim`` files if a matching ``.whl`` file already exists with the
+exact same file name tags. However, indexes **MAY** accept a ``.whl`` file if a matching
+``.rim`` file exists, as long as that ``.rim`` file hasn't been deleted or yanked. This
+allows uploaders to replace an externally hosted wheel file with an index hosted wheel
+file, but the converse is prohibited. Since the default is to host wheels on the same
+package index that contains the package metadata, it is not allowed to "downgrade" an
+existing wheel file once uploaded. When a ``.whl`` replaces a ``.rim``, the index **MUST**
+provide download URLs for the package using its own hosted file service. When uploading
+the overriding ``.whl`` file, the package index **MUST** validate the hash from the
+existing ``.rim`` file, and these hashes must match or the overriding upload **MUST** be
+rejected.
+
+PyPI API bump unnecessary
+=========================
```

Comment:
```suggestion
PyPI API bump unnecessary
-------------------------
```

Reply:
Thanks @AA-Turner.  Some of these sections do make sense under Specifications, but I'm not sure all do.